www/js/app.js

```diff
@@ -254,11 +254,33 @@ define(['jquery', 'abstractBackend', 'util', 'cookies','geometry','osabstraction
         $('#searchingForTitles').hide();
         return false;
     });
-    $('input:radio[name=contentInjectionMode]').on('change', function(e) {
+    $('input:radio[name=contentInjectionMode]').on('change', function(event) {
         // Do the necessary to enable or disable the Service Worker
         setContentInjectionMode(this.value);
         checkSelectedArchiveCompatibilityWithInjectionMode();
     });
+    $('#articleContent').on('load',onIFrameLoad);
+    function onIFrameLoad() {
+        $('#centeredspinner').hide();
+        $('#articleContent').contents().on('click', function (event) {
+            if (contentInjectionMode === 'serviceworker') {
+                // In Service Worker mode, we want to display a spinner
+                // when the user clicks on a link
+                // We only do that when the click is somewhere inside a <a> tag
```

Comment:
You're right : it's not a safe way at all. It was just an easy way to do a first version.
In fact, I would have expected the browser to inform the user by itself that a ServiceWorker is working. Just like it informs the user when some content is downloading.
It seems to be the case on Chromium, Firefox dev edition, but not on Firefox OS.

I should file a bug for Mozilla to implement it on Firefox OS, instead of trying to do it by hand.